README.md

```diff
@@ -9,3 +9,681 @@ ExpressRouterAdapter is available as a package on NPM for use with a module bund
+Yes, in fact this is one of the primary motivations behind express-route-adapter is to make that easy.
+
+In previous examples, we aren't specifying the media type, and so we're getting the default json formatter.
+
+Consider this example
+
+```js
+import { IHTTPRoute, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
+
+interface IPet {
+  name: string;
+  internalS3PathToPicture: string;
+}
+
+export class PetRouter {
+  pets: IPet[] = [
+    { name: 'honey', internalS3PathToPicture: '...' },
+    { name: 'poseidon', internalS3PathToPicture: '...' }
+  ];
+  getRoutes(): IHTTPRoute[] {
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .get(({ petId }) => {
+          return this.pets[petId];
+        })
+    ];
+  }
+}
+```
+
+Our data model has an internal path that we don't want to leak out.
+
+We can add our own media type to encapsulate this.
+
+```ts
+import { IHTTPRoute, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
+import { autoinject } from 'aurelia-dependency-injection';
+
+interface IPet {
+  name: string;
+  internalS3PathToPicture: string;
+}
+
+class PetMediaType {
+  mediaType: string = 'application/json';
+  formatForResponse({ name }: IPet): any {
+    return { name };
+  }
+}
+
+// tslint:disable-next-line: max-classes-per-file
+@autoinject
+export class PetRouter {
+  constructor(private petMediaType: PetMediaType){}
+  pets: IPet[] = [
+    { name: 'honey', internalS3PathToPicture: '...' },
+    { name: 'poseidon', internalS3PathToPicture: '...' }
+  ];
+  getRoutes(): IHTTPRoute[] {
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .mediaType(this.petMediaType)
+        .allowAnonymous()
+        .get(({ petId }) => {
+          return this.pets[petId];
+        })
+    ];
+  }
+}
+```
+
+But now lets imagine we want to make a breaking change, we want to split name into first name and last name. This is easy with media types. Our model will have both properties, and our media types can format correctly so that our old consumers still get name.
+
+```ts
+import { IHTTPRoute, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
+import { autoinject } from 'aurelia-dependency-injection';
+
+interface IPet {
+  firstName: string;
+  lastName: string;
+  internalS3PathToPicture: string;
+}
+
+class OldPetMediaType {
+  mediaType: string = 'application/json';
+  formatForResponse({ firstName, lastName }: IPet): any {
+    return { name: `${firstName} ${lastName}` };
+  }
+}
+
+// tslint:disable-next-line: max-classes-per-file
+class PetMediaType {
+  mediaType: string = 'application/pet+json';
+  formatForResponse({ firstName, lastName }: IPet): any {
+    return { firstName, lastName };
+  }
+}
+
+// tslint:disable-next-line: max-classes-per-file
+@autoinject
+export class PetRouter {
+  constructor(private petMediaType: PetMediaType, private oldPetMediaType: OldPetMediaType){}
```

Comment:
Break onto multiple lines so it doesnt get clipped when viewing readme

![image](https://user-images.githubusercontent.com/1964631/61544829-32365480-aa14-11e9-854b-939aa722485b.png)